src/sas/sascalc/dataloader/file_reader_base_class.py

```diff
@@ -157,33 +158,41 @@ def sort_one_d_data(self):
                 data.x_unit = self.format_unit(data.x_unit)
                 data.y_unit = self.format_unit(data.y_unit)
                 # Sort data by increasing x and remove 1st point
-                ind = np.lexsort((data.y, data.x))
-                data.x = np.asarray([data.x[i] for i in ind]).astype(np.float64)
-                data.y = np.asarray([data.y[i] for i in ind]).astype(np.float64)
+                self.ind = np.lexsort((data.y, data.x))
+                data.x = self.sort_1d_array(data.x)
+                data.y = self.sort_1d_array(data.y)
                 if data.dx is not None:
                     if len(data.dx) == 0:
                         data.dx = None
                         continue
-                    data.dx = np.asarray([data.dx[i] for i in ind]).astype(np.float64)
+                    data.dx = self.sort_1d_array(data.dx)
                 if data.dxl is not None:
-                    data.dxl = np.asarray([data.dxl[i] for i in ind]).astype(np.float64)
+                    data.dxl = self.sort_1d_array(data.dxl)
                 if data.dxw is not None:
-                    data.dxw = np.asarray([data.dxw[i] for i in ind]).astype(np.float64)
+                    data.dxw = self.sort_1d_array(data.dxw)
                 if data.dy is not None:
                     if len(data.dy) == 0:
                         data.dy = None
                         continue
-                    data.dy = np.asarray([data.dy[i] for i in ind]).astype(np.float64)
+                    data.dy = self.sort_1d_array(data.dy)
                 if data.lam is not None:
-                    data.lam = np.asarray([data.lam[i] for i in ind]).astype(np.float64)
+                    data.lam = self.sort_1d_array(data.lam)
                 if data.dlam is not None:
-                    data.dlam = np.asarray([data.dlam[i] for i in ind]).astype(np.float64)
+                    data.dlam = self.sort_1d_array(data.dlam)
                 if len(data.x) > 0:
                     data.xmin = np.min(data.x)
                     data.xmax = np.max(data.x)
                     data.ymin = np.min(data.y)
                     data.ymax = np.max(data.y)
 
+    def sort_1d_array(self, array=[]):
+        if array.any():
+            for i in self.ind:
+                if math.isnan(array[i]):
+                    array[i] = 0
+            array = np.asarray([array[i] for i in self.ind]).astype(np.float64)
+        return array
+
```

Comment:
1. Don't need a default value array.  What would it mean to sort an array without passing in an array?  If you do want an default value make it a tuple rather than a list so that lint doesn't complain about using a mutable object as the default.

2. Use numpy indexing to reorder the elements rather than list comprehension or for loop.  Similarly for replacing NaN with zero.

3. Can combine the conversion from list to array with the coercion from int to float and maybe save one pass through the list, depending on how numpy implements it.

4. Is sort_1d_array part of the public api for this class?  If not, add a leading underscore.

5. The sort_1d_array method doesn't actually sort.  "_reorder_1d_array" would be a better name.

6. Do you need to keep ind around?  If not, pass it as a parameter.  Lint will then complain that this doesn't need to be a method (it doesn't reference self), so you could make the method a @staticmethod or move it out of the class as a global function.

7. If data.dx contains NaN they will be pushed to the end of dx and replaced with 0., and dx will no longer be sorted.  Is this behaviour desired?

The following code addresses 1-6.
```
@staticmethod
def _reorder_1d_array(array, ind):
    array = np.asarray(array, dtype=np.float64)
    array[np.isnan(array)] = 0
    return array[ind]
```